### PR TITLE
[4] Au revoir JObject... 

### PR DESCRIPTION
**Pull request by PhilETaylor**
Manual replacement of all JObject to Joomla\CMS\Object\CMSObject references. 

ONE THING THAT NEEDS REAL TESTING is this change https://github.com/joomla/joomla-cms/compare/4.0-dev...PhilETaylor:byebye-jobject?expand=1#diff-b31b65d1210170b779c0e6e9c198290ddd7ea0be9e1be88f0d2d06f26ff263e4R110

Screenshot 2021-12-27 at 20 55 49

And YES I KNOW that `Joomla\CMS\Object\CMSObject` is already `@deprecated` but lets take it one step at a time...

### CODE REVIEW

File: administrator/components/com_finder/src/Controller/IndexerController.php [hunk 1]
```diff
@@ -261,9 +261,9 @@ public function optimize()
 	/**
 	 * Method to handle a send a \JSON response. The body parameter
 	 * can be an \Exception object for when an error has occurred or
-	 * a \JObject for a good response.
+	 * a CMSObject for a good response.
 	 *
-	 * @param   mixed  $data  \JObject on success, \Exception on error. [optional]
+	 * @param   \Joomla\CMS\Object\CMSObject|\Exception  $data  \JObject on success, \Exception on error. [optional]
```
Comment (brianteeman) [inline administrator/components/com_finder/src/Controller/IndexerController.php]:
```suggestion
	 * @param   \Joomla\CMS\Object\CMSObject|\Exception  $data  CMSObject on success, \Exception on error. [optional]
```

File: components/com_tags/src/View/Tag/HtmlView.php [hunk 2]
```diff
@@ -45,7 +46,8 @@ class HtmlView extends BaseHtmlView
 	/**
 	 * Tag data for the current tag or tags (on success, false on failure)
 	 *
-	 * @var    \JObject|boolean
+	 * @var    \Joomla\CMS\Object\CMSObject|boolean
```
Comment (wilsonge) [inline components/com_tags/src/View/Tag/HtmlView.php:49]:
```suggestion
	 * @var    \Joomla\CMS\Object\CMSObject|bool
```

**Comment by PhilETaylor** [comment]
I will fix the conflicts asap - serves me right for trying to do too much too soon

**Comment by wilsonge** [comment]
Yeah. I'm confident in the CMSObject::class change btw. We're doing it in various other places in core (e.g. https://github.com/joomla/joomla-cms/compare/4.0-dev...PhilETaylor:byebye-jobject?expand=1#diff-30fa9bc9ba82e382bf7ca46349946055a8f9f881d454dfd970971556828b7a6bR743)

**Comment by PhilETaylor** [comment]
After a very a scary rebase my mistake where I thought I had lost everything, I thankfully had not, and have forced pushed the changes after manually merging them.

**Comment by wilsonge** [comment]
Phew and thanks!

**Pull request merged**